core-bundle/src/Cron/LegacyCron.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Contao\CoreBundle\Cron;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Cron;
+use Contao\System;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+class LegacyCron implements ServiceAnnotationInterface
+{
+    /**
+     * @var ContaoFramework
+     */
+    protected $framework;
+
+    public function __construct(ContaoFramework $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    /**
+     * @Cron("minutely")
+     */
+    public function onMinutely(): void
+    {
+        $this->runLegacyCrons('minutely');
+    }
+
+    /**
+     * @Cron("hourly")
+     */
+    public function onHourly(): void
+    {
+        $this->runLegacyCrons('hourly');
+    }
+
+    /**
+     * @Cron("daily")
+     */
+    public function onDaily(): void
+    {
+        $this->runLegacyCrons('daily');
+    }
+
+    /**
+     * @Cron("weekly")
+     */
+    public function onWeekly(): void
+    {
+        $this->runLegacyCrons('weekly');
+    }
+
+    /**
+     * @Cron("monthly")
+     */
+    public function onMonthly(): void
+    {
+        $this->runLegacyCrons('monthly');
+    }
+
+    private function runLegacyCrons(string $interval): void
+    {
+        $this->framework->initialize();
+
+        if (isset($GLOBALS['TL_CRON'][$interval])) {
```

Comment:
Should we trigger a deprecation notice here?

Reply:
We can only do this once the core does not use `$GLOBALS['TL_CRON']` itself anymore.